packages/core/src/trpc.ts

```diff
@@ -71,18 +84,16 @@ export function trpcMiddleware(options: SentryTrpcMiddlewareOptions = {}) {
         }
 
         if (isThenable(maybePromiseResult)) {
-          return maybePromiseResult.then(
-            nextResult => {
-              captureIfError(nextResult);
-              span.end();
-              return nextResult;
-            },
-            e => {
-              captureException(e, trpcCaptureContext);
-              span.end();
-              throw e;
-            },
-          ) as T;
+          try {
+            const nextResult = await maybePromiseResult;
+            captureIfError(nextResult);
+            span.end();
+            return nextResult;
+          } catch (e) {
+            captureException(e, trpcCaptureContext);
+            span.end();
+            throw e;
+          }
```

Comment:
If we are gonna make the middleware return a promise we can simply await and try-catch the result instead of doing all of the maybePromiseResult gymnastics 👍